connectors/graphql/src/main/java/io/camunda/connector/graphql/GraphQLFunction.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.graphql;
+
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.gson.Gson;
+import io.camunda.connector.api.annotation.OutboundConnector;
+import io.camunda.connector.api.outbound.OutboundConnectorContext;
+import io.camunda.connector.api.outbound.OutboundConnectorFunction;
+import io.camunda.connector.graphql.auth.OAuthAuthentication;
+import io.camunda.connector.graphql.components.GsonComponentSupplier;
+import io.camunda.connector.graphql.components.HttpTransportComponentSupplier;
+import io.camunda.connector.graphql.model.GraphQLRequest;
+import io.camunda.connector.graphql.model.GraphQLResult;
+import io.camunda.connector.graphql.services.AuthenticationService;
+import io.camunda.connector.graphql.services.HTTPService;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@OutboundConnector(
+    name = "GRAPHQL",
+    inputVariables = {
+      "url",
+      "method",
+      "authentication",
+      "query",
+      "variables",
+      "connectionTimeoutInSeconds"
+    },
```

Comment:
A side comment: I think we might want to consider changing the naming pattern to `connectorName.variableName`, e.g. `graphql.url` here, because the current behavior of input mappings of variables with the same name is [sometimes surprising](https://github.com/camunda/connectors-bundle/issues/47).

Reply:
I didn't know about this, good catch. I will modify the input variables.